.circleci/config.yml

```diff
@@ -12,106 +20,156 @@ references:
+    steps:
+      - *cf_install
+      - *attach_workspace
+      - *cf_login_b_cld
+      - run: make -C api deploy-celery
+
   master_branch: &master_branch
     filters:
       branches:
-        only:
+        ignore:
           - master
   not_master_branch: &not_master_branch
     filters:
       branches:
-        ignore:
+        only:
           - master
 
 jobs:
   setup:
-    docker:
-      - image: circleci/python:3.6.5-node
+    <<: *python_node_image
     steps:
-        - checkout
-        - persist_to_workspace:
-            root: .
-            paths:
-              - Makefile
-              - utils
-              - api
-              - admin
+      - checkout
+      - persist_to_workspace:
+          root: .
+          paths:
+            - Makefile
+            - utils
+            - api
+            - admin
   build-admin:
-    docker:
-      - image: circleci/python:3.6.5-node
+    <<: *python_node_image
     steps:
-        - attach_workspace:
-            at: .
-        - run: make -C admin setup
-        - run: make -C admin vendor
-        - persist_to_workspace:
-            root: .
-            paths:
-              - admin
+      - *attach_workspace
+      - run: make -C admin setup
+      - run: make -C admin vendor
+      - persist_to_workspace:
+          root: .
+          paths:
+            - admin
   test-admin:
-    docker:
-      - image: circleci/python:3.6.5-node
+    <<: *python_node_image
     steps:
-        - attach_workspace:
-            at: .
-        - run: make -C admin setup-dev
-        - run: make -C admin test
+      - *attach_workspace
+      - run: make -C admin setup-dev
+      - run: make -C admin test
   build-api:
-    docker:
-      - image: circleci/python:3.6.5-node
+    <<: *python_node_image
     steps:
-        - attach_workspace:
-            at: .
-        - run: make -C api setup
-        - run: make -C api vendor
-        - persist_to_workspace:
-            root: .
-            paths:
-              - api
+      - *attach_workspace
+      - run: make -C api setup
+      - run: make -C api vendor
+      - persist_to_workspace:
+          root: .
+          paths:
+            - api
   deploy-dev-admin:
     environment:
+      - CLD_HOST: y.cld.gov.au
       - APP: notify-dev
-    <<: *deploy_admin
-  deploy-notify:
+    <<: *deploy_admin_y_cld
+  deploy-notify-y-cld:
     environment:
+      - CLD_HOST: y.cld.gov.au
       - APP: notify
-    <<: *deploy_admin
-  deploy-admin:
-    <<: *deploy_admin
-  deploy-api:
-    docker:
-      - image: circleci/python:3.6.5-node
-    steps:
-        - *cf_install
-        - attach_workspace:
-            at: .
-        - *cf_login
-        - run: make -C api deploy-api
-  deploy-celery:
-    docker:
-      - image: circleci/python:3.6.5-node
-    steps:
-        - *cf_install
-        - attach_workspace:
-            at: .
-        - *cf_login
-        - run: make -C api deploy-celery
+    <<: *deploy_admin_y_cld
```

Comment:
can we collapse these and put environment vars in the workspace part? 

like: 

* deploy defaults, -- deploy-api, deploy-admin, deploy-celery (staging)
* manual unlock
* set environment and re-perform deploys. deploy-api deploy-admin deploy-celery